### PR TITLE
User page refactor

### LINKED ISSUE #99: Tabs for different roles on user-list page

### PR DESCRIPTION
**Pull request by maksym-strus**
Added tabs for user-list page, closes #99

### CODE REVIEW

File: src/hooks/user/use-users-filtering.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const dispatch = useDispatch();
+
+  const handleTabChange = (tab) => {
+    if (!tab) {
+      setFilter((prevFilter) => ({
+        ...prevFilter,
+        roles: [user]
+      }));
+    } else {
+      setFilter((prevFilter) => ({
```
Comment (ecl1pseo) [inline src/hooks/user/use-users-filtering.js:20]: You duplicate here your code, you can use something like this:  `roles: tab ? [...other] : [user]`

File: src/pages/users/users-page/components/admin-tab/admin-tab.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const { CREATE_SPECIAL_USER } = config.buttonTitles;
+
+const AdminTab = (props) => {
+  const { list, onDelete } = props;
+  const styles = useStyles();
+  const dispatch = useDispatch();
+
+  const adminItems = list.map((userItem, index) => (
+    <TableContainerRow
+      key={index}
```
Comment (ecl1pseo) [inline src/pages/users/users-page/components/admin-tab/admin-tab.js:26]: Don't use index for key, use something stable like **id** or **name**

File: src/pages/users/users-page/components/user-tab/user-tab.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const tableHeaders = config.tableHeadRowTitles.users.userTab;
+
+const UserTab = (props) => {
+  const { list, onDelete } = props;
+  const dispatch = useDispatch();
+
+  const usersItems = list.map((userItem, index) => (
+    <TableContainerRow
+      key={index}
```
Comment (ecl1pseo) [inline src/pages/users/users-page/components/user-tab/user-tab.js:19]: Don't use index for key, use something stable like **id** or **name**

File: src/pages/users/users-page/components/user-tab/user-tab.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      id={userItem._id}
+      name={`${userItem.firstName} ${userItem.lastName}`}
+      mobile={formatPhoneNumber(userItem.phoneNumber)}
+      email={userItem.email}
+      role={userRoleTranslations[userItem.role]}
+      banned={userItem.banned ? 'Неактивний' : 'Активний'}
```
Comment (ecl1pseo) [inline src/pages/users/users-page/components/user-tab/user-tab.js:25]: Move this to translation file

File: src/pages/users/users-page/users-page.js [hunk 5]
```diff
@@ -47,47 +38,35 @@ const UsersPage = () => {
...
-      />
+      <AppBar position='static' color='primary'>
+        <Tabs
+          value={tabValue}
+          onChange={handleTabChange}
+          className={styles.tabs}
+          variant='fullWidth'
+        >
+          <Tab label='Користувачі' />
+          <Tab label='Адміністратори' />
```
Comment (ecl1pseo) [inline src/pages/users/users-page/users-page.js:61]: Move this to translation file

File: src/redux/users/users.saga.js [hunk 6]
```diff
@@ -45,13 +45,14 @@ const {
...
-function* handleUsersLoad() {
+function* handleUsersLoad({ payload }) {
   try {
     yield put(setUsersLoading(true));
-    const users = yield call(getAllUsers, null);
+    const users = yield call(getAllUsers, payload);
     yield put(setUsers(users));
     yield put(setUsersLoading(false));
   } catch (err) {
+    console.log(err);
```
Comment (ecl1pseo) [inline src/redux/users/users.saga.js:55]: Remove this `console.log`

File: src/pages/users/users-page/components/user-tab/user-tab.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      deleteHandler={() => onDelete(userItem._id)}
+      editHandler={() => dispatch(push(`/users/${userItem._id}`))}
+    />
+  ));
+
+  return (
+    <>
```
Comment (TorskyM) [inline src/pages/users/users-page/components/user-tab/user-tab.js:32]: You don't need to enclose a single `TableContainerGenerator` inside of a fragment

Comment (maksym-strus) [reply src/pages/users/users-page/components/user-tab/user-tab.js]: fixed

File: src/redux/users/users.saga.js (hunk 6 above)
Comment (maksym-strus) [reply src/redux/users/users.saga.js]: fixed

File: src/pages/users/users-page/users-page.js (hunk 5 above)
Comment (maksym-strus) [reply src/pages/users/users-page/users-page.js]: fixed

File: src/pages/users/users-page/components/user-tab/user-tab.js (hunk 4 above)
Comment (maksym-strus) [reply src/pages/users/users-page/components/user-tab/user-tab.js]: fixed

File: src/pages/users/users-page/components/user-tab/user-tab.js (hunk 3 above)
Comment (maksym-strus) [reply src/pages/users/users-page/components/user-tab/user-tab.js]: fixed

File: src/pages/users/users-page/components/admin-tab/admin-tab.js (hunk 2 above)
Comment (maksym-strus) [reply src/pages/users/users-page/components/admin-tab/admin-tab.js]: fixed

File: src/hooks/user/use-users-filtering.js (hunk 1 above)
Comment (maksym-strus) [reply src/hooks/user/use-users-filtering.js]: fixed

**Review by ecl1pseo — APPROVED** [review]

**Pull request merged**